src/tags/control/Number.js

```diff
@@ -72,80 +72,113 @@ const Model = types
       return isDefined(self.number);
     },
   }))
-  .actions(self => ({
-    getSelectedString() {
-      return self.number + ' star';
-    },
+  .actions(self => {
+    const Super = { validateValue: self.validateValue };
 
-    needsUpdate() {
-      if (self.result) self.number = self.result.mainValue;
-      else self.number = null;
-    },
+    return {
+      validateValue(value) {
+        if (!Super.validateValue(value)) return false;
+        if (!isDefined(value)) return true;
```

Comment:
should not those two lines be swapped?

Reply:
I think that it's better to put `Super`'s  method call at the beginning. It may be called in chain with some other mixing and that can become complicated.